include/hipSYCL/sycl/libkernel/detail/half_representation.hpp

```diff
@@ -125,6 +127,10 @@ inline half_storage create(double f) noexcept {
   return truncate_from(f);
 }
 
+inline half_storage create(int i) noexcept {
+  return i;
+}
```

Comment:
Do we really need changes in `half_storage`? Would it not be sufficient to have a `half(int x)` constructor that just does `fp16::create(static_cast<float>(x))`?

Reply:
True, we don't need it. I was making changes by blindly following the errors.